tests/robustness/model/replay.go

```diff
@@ -73,7 +73,7 @@ func (r *EtcdReplay) next() (request EtcdRequest, revision int64, index int) {
 	revision = r.eventHistory[r.eventHistoryIndex].Revision
 	index = r.eventHistoryIndex
 	operations := []EtcdOperation{}
-	for r.eventHistory[index].Revision == revision {
+	for index < len(r.eventHistory) && r.eventHistory[index].Revision == revision {
```

Comment:
```suggestion
	if index > len(r.eventHistory) {
		panic("Cannot progress further, exhausted whole event history")
	}
	for index < len(r.eventHistory) && r.eventHistory[index].Revision == revision {
```

Reply:
Coming into `next` I don't think we will ever hit that because of the loop condition in `StateForRevision`, when `index++` happens, we can go out of range.

Reply:
you are right. The `StateForRevision` code will correctly prevent overflowing index.

The only thing we need to fix is `next` function called on the last revision that incorrectly panic as it try to check if revision still continues under the next index.

Your fix is correct and doesn't require any changes. Thanks!

One request for next PR would be to provide more context to make review easier.